resources-master/online-shop/src/app/cart/cart.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit } from '@angular/core';
+import { CartService } from '../_services/cart.service';
+import { CartItem } from '../_models/cart-item.model';
+import { OrderService } from '../_services/order.service';
+
+
+@Component({
+  selector: 'app-cart',
+  templateUrl: './cart.component.html',
+  styleUrls: ['./cart.component.css']
+})
+export class CartComponent implements OnInit {
```

Comment:
This is a smart component, but it contains mark-up. I would prefer if you create another `SmartCartComponent`  which will contain the service calls (and the current `CartComponent` should just have inputs and outputs).